src/AudioTrackManager.ts

```diff
@@ -0,0 +1,93 @@
+import { AudioTrack } from "./types";
+import MediaAudio from "./MediaAudio";
+
+class AudioTrackManager {
+  public isPlaying: boolean;
+
+  private currentAudio: MediaAudio[];
+
+  constructor() {
+    this.isPlaying = false;
+    this.currentAudio = [];
+  }
+
+  public play(audioTracks: AudioTrack[]): void {
+    if (audioTracks.length === 0) return;
+
+    this.isPlaying = true;
+
+    audioTracks.forEach((audioTrack) => {
+      if (!audioTrack.url) return;
+
+      // Get the index of this current audio track if it exists.
+      const index = this.currentAudio.findIndex(
+        (currentAudio) => currentAudio.url === audioTrack.url,
+      );
+
+      // If the track already exists in currentAudio
+      if (index === -1) {
+        const audio = new MediaAudio(audioTrack.url);
+
+        audio.loop = audioTrack.loop;
+        audio.volume = audioTrack.volume;
+        audio.originalVolume = audio.volume;
```

Comment:
I'm confused, we already set 
`this.originalVolume = this.volume;`
in the MediaAudio constructor